### PR TITLE
Use a user agent library

### PR DESCRIPTION
**Pull request by ChrisJohnNOAA**
to detect bots
and to help with parsing/anonymization of user agent strings

### CODE REVIEW

File: logparse.py [hunk 1]
```diff
@@ -230,14 +231,9 @@ def filter_user_agents(self,
                            ):
         """Filter out requests from bots."""
```
Comment (callumrollo) [inline logparse.py:232]: remove the kwarg `bots=None` as no longer used

File: logparse.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from pathlib import Path
 import polars as pl
 from collections import Counter
+from user_agents import parse
```
Comment (callumrollo) [inline logparse.py:6]: add user_agents to env.yml pip section

**Review by callumrollo — CHANGES_REQUESTED** [review]
Looks good! I've tested it locally too. I've made a couple of small review requests

**Comment by ChrisJohnNOAA** [comment]
> Looks good! I've tested it locally too. I've made a couple of small review requests

Done

**Review by callumrollo — APPROVED** [review]

**Pull request merged**